kms/capi/ncrypt_windows.go

```diff
@@ -514,6 +539,57 @@ func cryptFindCertificateKeyProvInfo(certContext *windows.CertContext) error {
 	return nil
 }
 
+func cryptFindCertificatePrivateKey(certContext *windows.CertContext) (uintptr, error) {
+	var (
+		kh      windows.Handle
+		keySpec uint32
+		free    bool
+	)
+
+	if err := windows.CryptAcquireCertificatePrivateKey(certContext, windows.CRYPT_ACQUIRE_ONLY_NCRYPT_KEY_FLAG, nil, &kh, &keySpec, &free); err != nil {
+		return uintptr(0), err
+	}
+
+	return uintptr(kh), nil
+}
+
+func cryptFindCertificateKeyContainerName(certContext *windows.CertContext) (string, error) {
```

Comment:
@maraino This function is not currently used, and I think it will ultimately need to change a bit, but I'm leaving it it. Right now, it will populate the certificate context when the DLL calls succeed, and the error handling is now correct based on feedback.